pkg/network/domainspec/generators.go

```diff
@@ -254,42 +253,51 @@ type PasstLibvirtSpecGenerator struct {
 	vmi          *v1.VirtualMachineInstance
 }
 
+const (
+	passtLogFile      = "/var/run/kubevirt/passt.log" // #nosec G101
+	ifaceTypeUser     = "user"
+	ifaceBackendPasst = "passt"
+)
+
 func (b *PasstLibvirtSpecGenerator) Generate() error {
-	err := exec.Command("pgrep", "passt").Run()
-	if err == nil {
-		return fmt.Errorf("passt process is already running")
-	}
-	// remove passt interface from domain spec devices interfaces
-	foundDomainInterface := false
+	var passtIfaceFound bool
+	var passtIfaceIdx int
 	for i, iface := range b.domain.Spec.Devices.Interfaces {
 		if iface.Alias.GetName() == b.vmiSpecIface.Name {
-			b.domain.Spec.Devices.Interfaces = append(b.domain.Spec.Devices.Interfaces[:i], b.domain.Spec.Devices.Interfaces[i+1:]...)
-			foundDomainInterface = true
+			passtIfaceFound = true
+			passtIfaceIdx = i
 			break
 		}
 	}
-	if !foundDomainInterface {
+	if !passtIfaceFound {
 		return fmt.Errorf("failed to find interface %s in vmi spec", b.vmiSpecIface.Name)
 	}
 
-	ports := b.generatePorts()
-	args := append([]string{"--runas", "107", "-e"}, ports...)
-	cmd := exec.Command("/usr/bin/passt", args...)
-	cmd.SysProcAttr = &syscall.SysProcAttr{
-		AmbientCaps: []uintptr{unix.CAP_NET_BIND_SERVICE},
-	}
+	iface := b.generateInterface(&b.domain.Spec.Devices.Interfaces[passtIfaceIdx])
+	b.domain.Spec.Devices.Interfaces[passtIfaceIdx] = *iface
 
-	// connect passt's stderr to our own stdout in order to see the logs in the container logs
-	var reader io.ReadCloser
-	reader, err = cmd.StderrPipe()
-	if err != nil {
-		log.Log.Reason(err).Error("failed to get passt stderr")
-		return err
-	}
 	go func() {
+		for {
+			if _, err := os.Stat(passtLogFile); !errors.Is(err, os.ErrNotExist) {
+				break
+			}
+			time.Sleep(time.Second)
+		}
+
+		logFile, err := os.Open(passtLogFile)
```

Comment:
> Note that the for loop only checks if the log file exist, using os.Stat is more convenient for that purpose since there is no need to release resources.

I am not clear why it is more convenient. We can even argue about the logic being unsafe and even non-efficient.
- Between the time the file is is confirmed to exist, until the time it is opened, things can happen.
- If some other error exists, the `Open` operation is still executed.

If an error is returned due to `Open`, there is nothing to release.
So it seems to me that the same logic can be applied with `Open`, retrying until it succeeds or has an error that is not about a missing file.

There is another risk of this goroutine getting stuck forever on this. I am not sure about the implications of that.

Reply:
> If some other error exists, the Open operation is still executed.

Fixed

> If an error is returned due to Open, there is nothing to release.
So it seems to me that the same logic can be applied with Open, retrying until it succeeds or has an error that is not about a missing file.

If I understand correctly, you mean we will use `Open` in for loop, this is problematic because we will end up with having a defer call inside a for look which is not recommended.


> There is another risk of this goroutine getting stuck forever on this. I am not sure about the implications of that.

The log file should create by libvirt once the domain is started, if the domain hands or libvirt is broken, this gorutine will keep running, until virt-launcher pod is stops.
This can be optimized by using file watches.

This PR is about executing Passt with libvirt and not optimizing passt logging, with the futuristic passt network binding plugin we can get rid of this gorutine, at least it will be done in cleaner way.